### ISSUE TITLE
[FEATURE]: Setup React Skeleton Screens

### ISSUE THREAD

**Issue by Mayank0255** [Mayank0255/Stackoverflow-Clone#31]
Can watch this YouTube playlist for reference - [The Net Ninja](https://www.youtube.com/playlist?list=PL4cUxeGkcC9i6bZhMuAzQpC6YgLmB4k4-)

Set up the Skeletons for all relevant components like-
* __PostItem Component__
* __UserPanel Component__
* __TagPanel Component__
* __QuestionSection Component__
* __AnswerItem Component__
* __CommentCell Component__

**Comment by divyanshu887** [comment]
Hello sir @Mayank0255  i would like to work on this.

**Comment by Mayank0255** [comment]
@divyanshu887 

Sure, go ahead

**Comment by Mayank0255** [comment]
@divyanshu887 
May I ask for the progress? You can ask me if you are stuck anywhere

**Comment by divyanshu887** [comment]
> @divyanshu887
> May I ask for the progress? You can ask me if you are stuck anywhere

i am having mid sem exams till tomorrow so currently not much progess on this issue but i will try my best to send the pr  by the end of this week. I am beginner in frontend development so any guidance from you will be helpful . Sorry for the delay 
@Mayank0255

**Comment by Mayank0255** [comment]
@divyanshu887 
No worries, concentrate on your exam and all the best :)

For the beginner thing, I have attached a great learning source and it is what I want in my project too, so, take a look at it and of course I am here to guide as well

**Comment by divyanshu887** [comment]
@Mayank0255  i am unable to set up the project can you guide me. i dont understand what is the use of .env file

**Comment by Mayank0255** [comment]
@divyanshu887 

So, you have to setup .env file as it will be consisting of your mysql database credentials which will stay in your local machine and won't be going public to everyone. If that isn't there then you won't be able to connect with the database.

**Comment by divyanshu887** [comment]
@Mayank0255 to solve the issue , i have make a skeleton folder in src in which i will have to make skeleton of all the components mentioned above and import them to their jsx files .
is this process correct or not ?

**Comment by Mayank0255** [comment]
For now make the skeletons directory in the components directory and then proceed as you mentioned

**Comment by divyanshu887** [comment]
@Mayank0255  after creating env file , i am unable to find way to setup it locally. can you tell what to do next ? 

> @Mayank0255 i am unable to set up the project can you guide me. i dont understand what is the use of .env file

**Comment by Mayank0255** [comment]
@divyanshu887 Are you following the guidelines mentioned properly?

**Comment by Mayank0255** [comment]
Have filled your .env file exactly like .env.example? and have added the required credentials too?

**Comment by divyanshu887** [comment]
yes i have installed mysql and created a new database and filled it as described in env.example

**Comment by Mayank0255** [comment]
What error are you getting while running?

**Comment by divyanshu887** [comment]
i am getting this on running the command  written in readme in the directory where i have saved env file 

error code ENOENT
error syscall open
error path /home/imposter/react/package.json
error errno -2
error enoent ENOENT: no such file or directory, open '/home/imposter/react/package.json'
error enoent This is related to npm not being able to find a file.

**Comment by Mayank0255** [comment]
@divyanshu887 

So, this is not an .env error as right now you are just install deps.

1. Is the terminal set to the path of the directory? I not then do so and run again

2. If `npm run installDep` is not working then first run `npm install` and then `cd client` and then run `npm install`

**Comment by divyanshu887** [comment]
> @divyanshu887
> 
> So, this is not an .env error as right now you are just install deps.
> 
> I am assuming that you have node, npm and mysql installed in your pc properly and functioning properly too
> 
> 1. Is the terminal set to the path of the directory? I not then do so and run again
> 2. If `npm run installDep` is not working then first run `npm install` and then `cd client` and then run `npm install`

it worked

**Comment by divyanshu887** [comment]
@Mayank0255  getting this when trying to run 'npm run server'. please tell what i am doing wrong here
![Screenshot from 2021-03-25 22-05-24](https://user-images.githubusercontent.com/59568700/112509390-6d2a8700-8db6-11eb-9682-add6369e7cb1.png)

**Comment by Mayank0255** [comment]
You are doing setting up your .env wrong then send me ss of that.

P.S. - Is it the first time working on node and react?

**Comment by divyanshu887** [comment]
@Mayank0255  yes it is my first time working with react and node js. i have never used mysql so i am unable to set up the connection. i tried to solve the error but failed. i think first i should learn mysql and gain some experience in nodejs and react then i will be able to solve these issues .as i have not been able to solve the issue ,you can unassign me this issue for now, after learning the techstacks i will try to contribute in this project 
sorry for wasting your time 

![Screenshot from 2021-03-26 00-12-48](https://user-images.githubusercontent.com/59568700/112527905-88eb5880-8dc9-11eb-967b-b05cab9db293.png)

**Comment by Mayank0255** [comment]
@divyanshu887 Sure man, will be waiting for your contribution

**Comment by alceil** [comment]
Can i work on it?

**Comment by Rezanop** [comment]
Hello,
If @divyanshu887 isn't working in this issue anymore, i'd love to work on it!

**Comment by Mayank0255** [comment]
@Rezanop yes @divyanshu887 is not working on this issue anymore, should I assign it to you?

If yes then in how many days can I expect a PR?

**Comment by Rezanop** [comment]
I'll need to get familiar with skeleton api.
So I'm hoping to finish in a week or less if it's fine with you

**Comment by Mayank0255** [comment]
@Rezanop Yes that's fine with me, I just want contributors that will actually work on the project as some of them just take up issues and then forget about it.

I am assigning it to you then 🎉 

Let me know if you get stuck anywhere here on my discord Shadowsaver#9665.

**Comment by gyanendrasng** [comment]
is issue still there?

**Comment by Mayank0255** [comment]
@gyanendrasng yes, it is, would you be willing to work on this?

**Comment by hamzaiqbal2000** [comment]
is this issue still there?
